packages/server/src/wallet/services/blockchain.ts

```diff
@@ -1,16 +1,20 @@
 import { Address, Uint256 } from 'fmg-core';
 
+import * as AsyncLock from 'async-lock';
 import { nitroAdjudicator } from '../utilities/blockchain';
-
+const lock = new AsyncLock();
 export class Blockchain {
   static nitro: any;
   static async fund(channelID: Address, expectedHeld: Uint256, value: Uint256): Promise<Uint256> {
-    await Blockchain.attachNitro();
+    // We lock to prevent issues with the transaction nonce getting out of sync in ethers.js
```

Comment:
Can you explain how this kind of out-of-sync can arise? This kind of explanation is valuable to have in the commit message.

Reply:
I'll add a more detailed comment. Out of sync might not be the best wording but I I believe this is the same issue you ran into here (a while ago): https://github.com/ethers-io/ethers.js/issues/363 where sending multiple transactions around the same time results in the error `the tx doesn't have the correct nonce. account has nonce of: 95 tx has nonce of: 94`

With virtual funding it's relatively easy to run into this since both player A and player B can deposit around the same time which results in the hub trying to do two deposits at roughly the same time.

My solution to avoid running into this error was to block so only one deposit occurs at a time.